sensei/ConduitDataAdaptor.h

```diff
@@ -37,7 +37,7 @@ class SENSEI_EXPORT ConduitDataAdaptor : public sensei::DataAdaptor
 
 protected:
   ConduitDataAdaptor();
-  ~ConduitDataAdaptor();
+  virtual ~ConduitDataAdaptor();
```

Comment:
virtual is not needed here. the base class has virtual destructor, even without the virtual key word you've added here, this class does too. that's part of the language.

Suggestion: If you want to ensure that the code is correct or to clarify, instead use the override key word here both as a form of documentation and to have the compiler enforce that the base class has a virtual destructor, which is the important thing.

this applies elsewhere you've added virtual to a derived class's destructor declaration.